src/Files.App/Utils/Storage/Search/EverythingSearchEngineService.cs

```diff
@@ -0,0 +1,140 @@
+
+		public bool IsAvailable => _everythingService.IsEverythingAvailable();
+
+		public async Task<IList<ListedItem>> SearchAsync(string query, string? path, CancellationToken ct)
+		{
+			App.Logger?.LogInformation("[SearchEngine: Everything] Starting search - Query: '{Query}', Path: '{Path}'", query, path ?? "<global>");
+			
+			if (!IsAvailable)
+			{
+				App.Logger?.LogWarning("[SearchEngine: Everything] Everything search unavailable, performing fallback to Windows Search");
+				await NotifyEverythingUnavailableOnce();
+				return await FallbackToWindowsSearch(query, path, MaxSearchResults, ct);
+			}
+
+			try
+			{
+				var results = await _everythingService.SearchAsync(query, path, ct);
+				App.Logger?.LogInformation("[SearchEngine: Everything] Search completed - Found {ResultCount} results", results.Count);
+				return results;
+			}
+			catch (Exception ex)
+			{
+				App.Logger?.LogError(ex, "[SearchEngine: Everything] Search failed, falling back to Windows Search");
+				return await FallbackToWindowsSearch(query, path, MaxSearchResults, ct);
+			}
+		}
+
+		public async Task<IList<ListedItem>> SuggestAsync(string query, string? path, CancellationToken ct)
+		{
+			App.Logger?.LogInformation("[SearchEngine: Everything] Starting suggestions - Query: '{Query}', Path: '{Path}'", query, path ?? "<global>");
+			
+			if (!IsAvailable)
+			{
+				App.Logger?.LogWarning("[SearchEngine: Everything] Everything search unavailable, performing fallback to Windows Search for suggestions");
+				await NotifyEverythingUnavailableOnce();
+				return await FallbackToWindowsSearch(query, path, MaxSuggestionResults, ct);
+			}
+
+			try
+			{
+				// Use Everything API with limited results for suggestions
+				var results = await _everythingService.SearchAsync(query, path, ct);
+				// Limit to suggestion count
+				var limitedResults = results.Take(MaxSuggestionResults).ToList();
+				App.Logger?.LogInformation("[SearchEngine: Everything] Suggestions completed - Found {ResultCount} results", limitedResults.Count);
+				return limitedResults;
+			}
+			catch (Exception ex)
+			{
+				App.Logger?.LogError(ex, "[SearchEngine: Everything] Suggestions failed, falling back to Windows Search");
+				return await FallbackToWindowsSearch(query, path, MaxSuggestionResults, ct);
+			}
+		}
+
+		/// <summary>
+		/// Notifies user once per session that Everything is unavailable and Windows Search fallback is being used
+		/// </summary>
+		private async Task NotifyEverythingUnavailableOnce()
+		{
+			lock (_notificationLock)
+			{
+				if (_hasNotifiedEverythingUnavailable)
+					return;
+					
+				_hasNotifiedEverythingUnavailable = true;
+			}
+			
+			try
+			{
+				App.Logger?.LogInformation("[SearchEngine: Everything] Showing fallback notification to user");
+				
+				// Show toast notification that Everything is unavailable
+				await Task.Run(() =>
+				{
+					AppToastNotificationHelper.ShowEverythingUnavailableToast();
+				});
```

Comment:
This block the app from building for me

<img width="1150" height="81" alt="Image" src="https://github.com/user-attachments/assets/8736172a-ea7d-4dcb-9875-9ee3dbcfe517" />